### DISCUSSION TITLE
Help with my onboarding migration guide

### DISCUSSION THREAD

**Discussion by bitsondatadev** [immich-app/immich#12369] [General]
Hello Immich nation,

I am brand new to the community and am currently in the process of migration. I notice there's a lot of scattered information around onboarding into Immich along with a lot of repeat questions. While the community has done well to help on a case by case basis, I would like to help define a "recommended" approach that covers most use cases and ideally becomes an official guide in the documentation as a way to get started.

Everyone has a bit of a different journey and requirements, for example, [this user coming from flickr](https://github.com/immich-app/immich/discussions/12351), but there are absolutely common tasks that could be made simpler. Also, having sections that touch on recommended approaches based on shared experience for different data sources would also be valuable.

I wrote up [this community migration guide](https://github.com/immich-app/immich/discussions/12334) which currently detailed my steps so far with Google and adds links to previous guides and tools with my notes. For those who have contributed documentation, and any contributors that have interest in optimizing onboarding workflow please take a look.

Specific questions:

 - Are there any errata?
 - What am I missing?
 - Is this too prescriptive?

Link: https://github.com/immich-app/immich/discussions/12334

Tagging a few that I see contribute to docs: @alextran1502, @jrasm91, @mmomjian, @aviv926, @ppr88 Thanks!

**Comment by bitsondatadev** [comment]
Another thread:

What are folks' thoughts on this opinion?

> Since the cli has kept up closely with the go features and [fixed original limitations from when immich-go was created](https://github.com/simulot/immich-go/tree/385c9c528c0f8f2d6c30d6c9767d78866716cd5d?tab=readme-ov-file#where-the-immich-cli-falls-short), having both immich-go and immich-cli tends to confuse new users. Ideally immich-go maintainer can merge their efforts with core immich team, potentially providing Docker/Go binary). This guide will aim to recommend immich supported options where they exist and just using Docker/Podman to generalize the toolset.

@simulot I'm curious on your take here. I think the work you're doing is incredible. That said, from a newcomers perspective, it seems like it may be a somewhat duplicated effort. At the very least, are there possible plans on you joining the Immich team or somehow syncing your work with theirs?

Having a go version of the cli is great if they both share the same features, but at this point it's yet another decision I have to make and learn about.

**Reply by simulot** [reply]
Given my limited proficiency in TypeScript and JavaScript, I'm unable to implement the Google Photos feature in the official CLI. Sorry.

**Reply by bitsondatadev** [reply]
> Given my limited proficiency in TypeScript and JavaScript, I'm unable to implement the Google Photos feature in the official CLI. Sorry.
> 
> 

I imagine this would be someone else porting features over if that's something you're okay with. Or at least an effort to keep functionality aligned so that it offloads that choice. 

That said, the official wisdom is actually to use your library so I'll update the guide to do that.

**Comment by alextran1502** [comment]
Cc @mmomjian and @aviv926 since you guys are familiar with documentation, any thoughts on this?

**Comment by aviv926** [comment]
It looks good, I agree that in the end the ultimate goal is to create one road that will be the "King's Road" for a smooth transition to Immich.

A few notes:
* For everything related to importing from Google Photos (Takeout), I think there are 2 tools today that are the best and require almost 0 understanding on the part of the basic user:

  https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper
  This tool is supported on all platforms (also Android with Tremux but I wouldn't recommend trying...lol.)
  
  This tool is an excellent guide in itself, within a few minutes it includes everything you *really* need to get a clean, tidy copy of all the photos from Takeout
  
  It can be an alternative to all the tools mentioned in the manual in phase 2.
  
  In the future if #9804 is merged it will solve all the problem related to Google Photos
  
  I also saw that @bo0tzz mentioned [dtinit](https://dtinit.org/) it could be great for community maintenance to have everything in one place.

* As for Immich-go, I think it's the best tool for merging from Google Images, I agree that it would be great to merge this tool with Immich-CLI because it seems that in the end both pretend to be the same, it could be that the limitation of merging comes from the language in which they are written.

* As for Apple, I thought of an option that could solve a lot of problems, it comes at the price of privacy, but unfortunately in 2024 this is the way /:
The idea is to use the official transfer tool from Apple to Google Photos and then when everything is in Google Photos to takeout everything and import with Immich-go, this is the fastest and best solution for all cloud solutions at this point.

  Google has a support page on this:
  https://support.google.com/photos/answer/10502587

  > Google supports the transfer of your information from Google Photos to:
  >   Supported third party services: Apple – iCloud Photos, Flickr, Microsoft One Drive and SmugMug.

  and vice versa
  https://support.apple.com/en-us/120924

* For the matter of Automatic creation of albums in external libraries
This is an option available with the new Folder view feature, although it does not create an album but displays it.

  You can create an album from an external library using a script that someone from the community created:
  https://github.com/alvistar/immich-albums
  I haven't tried, and I'm not sure how up to date it is.

**Reply by bitsondatadev** [reply]
This is incredibly helpful @aviv926! I'll update this and also keep working on finding a better solution for Apple (Since I kind of have to :) ). Though, @alextran1502 mentioned to me that doing it from your phone should work fine and still maintain exif and other data.

Obviously this could be a tedious process if you don't have a lot of storage on your phone, but it's _something_.

I'll also slowly try to map and connect as many issues/prs/discussions that I can to lower the research needed by the core team when considering how to make this process better.

I think having everything organized and then having some debate about what has the best cost/benefit value for this project and when it fits in the larger narrative of other Immich features.

Anyways, thanks again. I really look forward to pushing things forward here.

**Reply by aviv926** [reply]
> This is incredibly helpful @aviv926! I'll update this and also keep working on finding a better solution for Apple (Since I kind of have to :) ). Though, @alextran1502 mentioned to me that doing it from your phone should work fine and still maintain exif and other data.
> 
> Obviously this could be a tedious process if you don't have a lot of storage on your phone, but it's _something_.
> 
> I'll also slowly try to map and connect as many issues/prs/discussions that I can to lower the research needed by the core team when considering how to make this process better.
> 
> I think having everything organized and then having some debate about what has the best cost/benefit value for this project and when it fits in the larger narrative of other Immich features.
> 
> Anyways, thanks again. I really look forward to pushing things forward here.

I don't have Apple services, but isn't exporting from Apple and uploading to Immich something that works well?

**Reply by bitsondatadev** [reply]
I'll have to verify myself. I've only migrated google stuff out. At the very least having something in a guide that says, "migrating from Apple? don't sweat it!" would be enough just so users are aware they don't have to take the same measures.

**Comment by mmomjian** [comment]
Thanks for this. It's not clear to me if the goal of this is 1) just remain a community discussion guide, 2) be linked as a Community Guide in the docs, or 3) merged into the mainline docs itself? If 1, you can pretty much write whatever you'd like! If 2 or 3, I don't think it's currently in a position that we would list it as a "recommended" resource. A few reasons I say that:

- We do not recommend Podman over Docker Compose. It's harder to setup for limited benefit to the average user
- I don't know of anyone here using google-photos-migrate. Our recommend pathway for Google Takeout is immich-go, as mentioned later
- Style of writing is not consistent with the Immich docs

Again, if this remains a discussion thread, it's fine, but I wanted to give some idea. Thanks again for writing it up!

**Reply by bitsondatadev** [reply]
Thanks @mmomjian! I want this to start out as a discussion open for newcomers like me to have a place to start from when migrating to Immich (initially option 1).

I'll eventually (perhaps sooner than later) want to take steps needed to get this integrated in the docs. Would you be able to point me to a style guide that you use for the Immich docs so I can work on cleaning this up at some point (e.g. https://developers.google.com/style).

Thanks!

**Reply by bitsondatadev** [reply]
Also, thanks about the immich-go info...I'll incorporate that into the migration guide instead and avoid pushing Podman. Are there any plans of Immich adopting immich-go features into immich-cli in the near-ish future (Q1/Q2 2025?) I ask because that seems to be the clearest way to simplify the migration is to remove the question of which tool to use for pre-processing, while simultaneously moving toward merging the metadata and pre-processing steps.

That is just my opinion.
